themes/doks/layouts/shortcodes/figure.html

```diff
@@ -22,9 +22,10 @@
   {{ $src = .Get 0 }}
   {{ $alt = .Get 1 }}
 {{end}}
+{{ $image := .Page.Resources.GetMatch (printf "*%s*" (.Get "src")) -}}
 <figure>
   {{if $zoom}}<a href="{{$zoom}}" target="_blank">{{end}}
-  <img src="{{$src}}" alt="{{$alt}}"{{if $title  }} title="{{$title}}"{{end}}{{if $width}} width="{{$width}}"{{end}}{{if $height }} height="{{$height}}"{{end}}{{if $align}} align="{{$align}}"{{end}}/>
+  <img {{if $image }} src="{{ $image.RelPermalink }}"{{ else }}src="{{$src}}" {{ end }} alt="{{$alt}}"{{if $title }} title="{{$title}}"{{end}}{{if $width}} width="{{$width}}"{{end}}{{if $height }} height="{{$height}}"{{end}}{{if $align}} align="{{$align}}"{{end}}/>
```

Comment:
```suggestion
  <img {{if $image }} src="{{ $image.RelPermalink }}" {{ else }} src="{{$src}}" {{ end }} alt="{{$alt}}" {{if $title }} title="{{$title}}" {{end}} {{if $width}} width="{{$width}}" {{end}} {{if $height }} height="{{$height}}" {{end}} {{if $align}} align="{{$align}}" {{end}}/>
```
Or actually use line breaks:
```suggestion
  <img
            {{if $image }}
                src="{{ $image.RelPermalink }}"
            {{ else }}
                src="{{$src}}" 
            {{ end }}
            alt="{{$alt}}"
            {{if $title }} 
                title="{{$title}}"
            {{end}}
            {{if $width}}
                width="{{$width}}"
            {{end}}
            {{if $height }}
                height="{{$height}}"
            {{end}}
            {{if $align}}
                align="{{$align}}"
            {{end}}
        />
```

Reply:
Thanks. I've committed your suggestion. I was just trying to quickly facilitate the preview of the new structure rather than improve the existing code.